packages/subgraph/README.md

```diff
@@ -98,6 +98,112 @@ To stop:
+
+```js
+export class ExtendedTokenVotingProposal extends TokenVotingProposal {
+  withDefaultValues(): ExtendedTokenVotingProposal {
+    this.id = PROPOSAL_ENTITY_ID;
+    this.dao = DAO_ADDRESS;
+    this.plugin = Address.fromHexString(CONTRACT_ADDRESS).toHexString();
+    this.pluginProposalId = BigInt.fromString(PLUGIN_PROPOSAL_ID);
+    this.creator = Address.fromHexString(ADDRESS_ONE);
+    this.open = true;
+    this.executed = false;
+    this.votingMode = VOTING_MODE;
+    /// some more properties
+  }
+
+  createEvent_ProposalCreated(
+    actions: ethereum.Tuple[],
+    description: string = STRING_DATA
+  ): ProposalCreated {
+    let event = createNewProposalCreatedEvent(
+      this.pluginProposalId.toString(),
+      this.creator.toHexString(),
+      this.startDate.toString(),
+      this.endDate.toString(),
+      description,
+      actions,
+      this.allowFailureMap.toString(),
+      this.plugin
+    );
+    return event;
+  }
+
+  assertEntity(debug: boolean = false): void {
+    let entity = TokenVotingProposal.load(this.id);
+    if (!entity) throw new Error('Entity not found');
+    let entries = entity.entries;
+    for (let i = 0; i < entries.length; i++) {
+      let key = entries[i].key;
+
+      let value = this.get(key);
+
+      assert.fieldEquals('TokenVotingProposal', this.id, key, value.displayData());
+    }
+  }
+}
+```
+
+There're a couple of things we achieved:
+
+- The default values are not scattered anymore and only live in their appropire extended entity classes.
+- There's no need to call `fieldEquals` multiple times for each field in the tests. Simply calling `assertEntity` is enough.
+- Creating the events is much simpler, because it uses the same default values that live in the same class which brings much more safety.
+
+Usage of the extended class is the following:
+
+```js
+let proposal = new ExtendedTokenVotingProposal().withDefaultValues();
+let event = proposal.createEvent_ProposalCreated(actions, STRING_DATA);
+_handleProposalCreated(event, proposal.dao, STRING_DATA);
+// If the event uses non-default values for a couple of fields
+// Update the proposal's fields with those values
+proposal.creationBlockNumber = BigInt.fromString(ONE);
+proposal.votingMode = VOTING_MODES.get(parseInt(VOTING_MODE)) as string;
+// assert
+proposal.assertEntity();
+```
+
+**How it works behind the hood**
+
+When you have a new entity in the schema, go to `/helpers/method-classes.ts` and add a new class that extends from that entity. Then add the functions that you want this extended class to have.
+
+After running `yarn test`, our script('schema-extender.ts') loops through the classes in method-classes and cleverly generates the `helpers/extended-schema.ts` which can be used in the tests to use Extended classes instead of the original entity classes.
+
```

Comment:
```suggestion

Setting properties on the extended entity (before saving) will allow you to run `assertEntity` to check that the properties _fetched from the database_ match the properties you've just set, else it will use the default values.

In the above example we are asserting that the `creationsBlockNumber` and `votingMode` are set to specific values, while the rest of the entity will be checked against the defaults.
```